router/path_hosted.go

```diff
@@ -0,0 +1,22 @@
+// +build hosted
+
+package router
+
+import (
+	"strings"
+)
+
+// extractPath extracts path from hosted EG host name (<space>.eventgateway([a-z-]*)?.io|slsgateway.com)
+func extractPath(host, path string) string {
+	subdomain := strings.Split(host, ".")[0]
+	return "/" + subdomain + path
+}
+
+func systemPathFromSpace(space string) string {
+	return "/" + space + "/"
+}
+
+// systemPathFromPath constructs path from path on which event was emitted. Helpful for "event.received" system event.
+func systemPathFromPath(path string) string {
+	return "/" + strings.Split(path, "/")[1] + "/"
```

Comment:
We guarantee here somehow that `path` would never be `/` only, correct? Otherwise this will blow up during subindexing.

Reply:
Yes, this code is only compiled for hosted EG. We guarantee that the first segment of path is always space name. Also if that assumption is wrong it's actually better to blow up asap :)